src/Mod/PartDesign/Gui/TaskHoleParameters.cpp

```diff
@@ -932,21 +955,14 @@ void TaskHoleParameters::changedObject(const App::Document&, const App::Property
         ui->Depth->setDisabled(ro);
     }
     else if (&Prop == &hole->DrillPoint) {
-        ui->drillPointFlat->setEnabled(true);
-        ui->drillPointAngled->setEnabled(true);
         std::string drillPoint(hole->DrillPoint.getValueAsString());
-        if (drillPoint == "Flat" && !ui->drillPointFlat->isChecked()) {
-            ui->drillPointFlat->blockSignals(true);
-            ui->drillPointFlat->setChecked(true);
-            ui->drillPointFlat->blockSignals(false);
-        }
-        if (drillPoint == "Angled" && !ui->drillPointAngled->isChecked()) {
-            ui->drillPointAngled->blockSignals(true);
-            ui->drillPointAngled->setChecked(true);
-            ui->drillPointAngled->blockSignals(false);
+        ui->DrillPointAngled->setEnabled(true);
+        if (ui->DrillPointAngled->isChecked() ^ (drillPoint == "Angled")) {
+            ui->DrillPointAngled->blockSignals(true);
+            ui->DrillPointAngled->setChecked(drillPoint == "Angled");
+            ui->DrillPointAngled->blockSignals(false);

```

Comment:
instead of manually blocking signals you should use `QSignalBlocker blocker(ui->DrillPointAngled)`